File: src/test/unit/LectureTest.java
package test.unit;
import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

import IDs.*;
import Enums.*;
import person.*;
import lecture.Lecture;
import lecture.LectureSession;

public class LectureTest {
	
	@Test
	public void test_Lecture_default_constructor() {
		LectureID testLecID1 = new LectureID("000");
		LectureID testLecID2= new LectureID("001");
		LectureID testLecID3= new LectureID("002");
		
		SessionID testSemID1 = new SessionID(23);
		SessionID testSemID2 = new SessionID(24);
		SessionID testSemID3 = new SessionID(25);
		
		LectureHour[][] sessionHours = new LectureHour[7][10];
		for (int i = 0; i<7; i++) {
			for(int j = 0; j<10; j++) {
				sessionHours[i][j] = LectureHour.NO;
			}
		}
		sessionHours[2][3] = LectureHour.YES;
		InstructorID testInsID1 = new InstructorID(123,123);
		InstructorID testInsID2 = new InstructorID(124,124);
		Instructor testInstructor1 = new Instructor("testInsFName1","testInsLName1",testInsID1,
				null,null,InstructorType.Instructor);
		Instructor testInstructor2 = new Instructor("testInsFName2","testInsLName2",testInsID2,
				null,null,InstructorType.Assistant);
		
		LectureSession testSession1 = new LectureSession(testSemID1,null,sessionHours,SessionType.Application,
				testInstructor1,null);
		LectureSession testSession2 = new LectureSession(testSemID2,null,sessionHours,SessionType.Theorytical,
				testInstructor1,null);
		LectureSession testSession3 = new LectureSession(testSemID3,null,sessionHours,SessionType.Theorytical,
				testInstructor1,null);
		
		List<LectureSession> testSessions = new ArrayList<>();
		testSessions.add(testSession1);
		testSessions.add(testSession2);
		
		List<LectureSession> testSessions2 = new ArrayList<>();
		testSessions.add(testSession1);
		testSessions.add(testSession3);
		
		List<Instructor> testListAssistants = new ArrayList<>();
		testListAssistants.add(testInstructor2);
		
		Lecture testLecture1 = new Lecture(testLecID1,"testLecName1", LectureType.FTE, 3,
				testSessions, null, 5);
		Lecture testLecture2 = new Lecture(testLecID3,"testLecName3", LectureType.MANDATORY, 2,
				testSessions, null, 5);
		Lecture testLecture = new Lecture(testLecID2,"testLecName2", LectureType.NTE, 4,
				testSessions, testLecture1, 6);
		
		testSession1.setLecture(testLecture1);
		testSession2.setLecture(testLecture);
		
		testSession1.setListOfAssistans(testListAssistants);
		testSession2.setListOfAssistans(testListAssistants);
		
		testLecture.setId(testLecID1);
		assertEquals(testLecID1, testLecture.getId());
		
		testLecture.setCredit(4);
		assertEquals(4, testLecture.getCredit());
		
		testLecture.setLectureType(LectureType.MANDATORY);
		assertEquals(LectureType.MANDATORY, testLecture.getLectureType());
		
		testLecture.setName("testString");
		assertEquals("testString", testLecture.getName());
		
		testLecture.setQuota(23);
		assertEquals(23, testLecture.getName());
		
		testLecture.setSessions(testSessions2);
		assertEquals(testSessions2, testLecture.getSessions());
		
		testLecture.setPrerequisite(testLecture2);
		assertEquals(testLecture2, testLecture.getPrerequisite());
	}

}

File: pythonCode/LectureTest.py
#!/usr/bin/env python
import unittest
from Lecture import Lecture
from LectureID import LectureID
from LectureType import LectureType
from LectureSession import LectureSession
class LectureTest(unittest.TestCase):
    # generated source for class LectureTest
    def test_lecture_lectureID(self):
        # generated source for method test_lecture_lectureID 
        testLecture = Lecture()
        testLectureID = LectureID("000")
        testLecture.setId(testLecture)
        self.assertEqual(testLectureID, testLecture.getID())
    def test_lecture_Name(self):
        # generated source for method test_lecture_Name 
        testLecture = Lecture()
        testLecture.setName("testString")
        self.assertEqual("testString", testLecture.getName())
    def test_lecture_LectureType(self):
        # generated source for method test_lecture_LectureType 
        testLecture = Lecture()
        testLecture.setLectureType(LectureType.MANDATORY)
        self.assertEqual(LectureType.MANDATORY, testLecture.getLectureType())
    def test_lecture_Credit(self):
        # generated source for method test_lecture_Credit 
        testLecture = Lecture()
        testLecture.setCredit(4)
        self.assertEqual(4, testLecture.getCredit())
    def test_lecture_Sessions(self):
        # generated source for method test_lecture_Sessions 
        testLecture = Lecture()
        testLectureSession = LectureSession()
        testLectureSession.setLecture(testLecture)
        testLecture.setSessions(testLectureSession)
        self.assertEqual(testLectureSession, testLecture.getSessions())
    def test_lecture_Prerequisite(self):
        # generated source for method test_lecture_Prerequisite 
        testLecture1 = Lecture()
        testLecture2 = Lecture()
        testLecture2.setName("testLName2")
        testLecture1.setPrerequisite(testLecture2)
        self.assertEqual(testLecture2, testLecture1.getPrerequisite())
if __name__ == '__main__':
    unittest.main()